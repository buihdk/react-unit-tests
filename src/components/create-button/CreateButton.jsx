import React, { memo } from 'react';
import PropTypes from 'prop-types';
import I18n from 'i18n-js';

import { Button } from 'antd';
import { HomeOutlined } from '@ant-design/icons';

import { noop } from 'src/utils';

export const CreateButton = ({ className, onClick, label }) => (
  <Button type="primary" onClick={onClick} className={className}>
    <HomeOutlined />
    {I18n.t(label)}
  </Button>
);

CreateButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string.isRequired,
};

CreateButton.defaultProps = {
  className: '',
  onClick: noop,
};

export default memo(CreateButton);
