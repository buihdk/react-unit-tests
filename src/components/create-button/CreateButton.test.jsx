import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CreateButton } from './CreateButton';

describe(`CreateButton`, () => {
  test(`renders without crashing`, () => {
    const props = {
      onShowModal: () => {},
      className: 'button',
      label: 'campaigns.create',
    };
    const wrapper = shallow(<CreateButton {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
