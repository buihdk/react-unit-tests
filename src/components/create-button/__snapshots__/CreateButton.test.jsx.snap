// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateButton renders without crashing 1`] = `
<Button
  block={false}
  className="button"
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  type="primary"
>
  <ForwardRef(HomeOutlined) />
  Create a New Campaign
</Button>
`;
